### PR TITLE
fix: Tooltip not hidden when menu is in control mode

### LINKED ISSUE #16742: 当 Menu 的 collapsed 处于受控状态时，且 collapsed 的改变是由 Menu 的 click 引起的，那么会引起 tooltip 非正常出现
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/lingering-sound-og3js)

### Steps to reproduce
进入 codesandbox ( https://codesandbox.io/s/lingering-sound-og3js )
- 点击左边菜单 nav1，使得菜单展开
- 点击 Hide 按钮，使得菜单收回
- 点击左边菜单 nav3，
- 点击 Hide 按钮，使得菜单再次收回

### What is expected?
菜单上没有 Tooltip 显现

### What is actually happening?
菜单上有 Tooltip 显现(甚至有两个)

image

| Environment | Info |
|---|---|
| antd | 3.18.2 |
| React | 16.8 |
| System | MacOS |
| Browser | Chrome  74.0.3729.157 (Official Build) (64-bit) |

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

fix #16742

### 💡 Solution

Reset Tooltip visible when disabled

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Tooltip not hidden when Menu collapsed in control mode |
| 🇨🇳 Chinese | 修复 Menu 在受控模式下收起 Tooltip 不消失的问题  |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

PS: Not easy way to add test case

### CODE REVIEW

File: components/menu/MenuItem.tsx [hunk 1]
```diff
@@ -37,13 +37,19 @@ export default class MenuItem extends React.Component<MenuItemProps> {
...
       <MenuContext.Consumer>
         {({ inlineCollapsed }: MenuContextProps) => {
+          const tooltipProps: TooltipProps = {};
+
           let titleNode = title || (level === 1 ? children : '');
           if (!siderCollapsed && !inlineCollapsed) {
             titleNode = null;
+            // Reset `visible` to fix control mode tooltip display not correct
+            // ref: https://github.com/ant-design/ant-design/issues/16742
+            tooltipProps.visible = false;
```
Comment (afc163) [inline components/menu/MenuItem.tsx]: 这样会不会导致 collapsed 的时候，Tooltip 出不来了？

Comment (zombieJ) [reply components/menu/MenuItem.tsx]: 不会，这个逻辑只有在展开情况下才会执行到。

**Comment by afc163** [comment]
https://circleci.com/gh/ant-design/ant-design/83836?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

```


node_modules/antd/lib/mentions/index.d.ts:25:51 - error TS2307: Cannot find module '_rc-mentions@0.3.1@rc-mentions/lib/Option'.

25     static Option: React.FunctionComponent<import("_rc-mentions@0.3.1@rc-mentions/lib/Option").OptionProps>;
                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/antd/lib/mentions/index.d.ts:34:50 - error TS2307: Cannot find module '_rc-mentions@0.3.1@rc-mentions/lib/util'.

34     getFilterOption: () => false | typeof import("_rc-mentions@0.3.1@rc-mentions/lib/util").filterOption | undefined;
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by afc163** [comment]
https://circleci.com/gh/ant-design/ant-design/83845?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

```

Summary of all failing tests
 FAIL  components/mentions/__tests__/index.test.js
  ● Mentions › loading

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 0

      71 |     const wrapper = mount(<Mentions loading />);
      72 |     simulateInput(wrapper, '@');
    > 73 |     expect(wrapper.find('.ant-mentions-dropdown-menu-item').length).toBe(1);
         |                                                                     ^
      74 |     expect(wrapper.find('.ant-spin').length).toBeTruthy();
      75 |   });
      76 | 

      at Object.toBe (components/mentions/__tests__/index.test.js:73:69)

  ● Mentions › notFoundContent

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 0

      78 |     const wrapper = mount(<Mentions notFoundContent={<span className="bamboo-light" />} />);
      79 |     simulateInput(wrapper, '@');
    > 80 |     expect(wrapper.find('.ant-mentions-dropdown-menu-item').length).toBe(1);
         |                                                                     ^
      81 |     expect(wrapper.find('.bamboo-light').length).toBeTruthy();
      82 |   });
      83 | });

      at Object.toBe (components/mentions/__tests__/index.test.js:80:69)


Test Suites: 1 failed, 153 passed, 154 total
Tests:       2 failed, 3 skipped, 1377 passed, 1382 total
```

**Pull request merged**